Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,711 @@
+  {
+    //  Remove currentRow from list
+    var selectedRows = BocList._selectedRows[bocList.id]!;
+    selectedRows.Rows[rowBlock.SelectorControl.id] = null;
+    selectedRows.Length--;
+
+    // Unselect currentRow
+    rowBlock.SelectorControl.checked = false;
+    if (isRowHighlightingEnabled)
+      $(rowBlock.Row).removeClass(BocList.TrClassNameSelected);
+
+    BocList.ClearSelectAllRowsSelector (selectedRows);
+  }
+
+  private static SetSelectAllRowsSelectorOnDemand (selectedRows: BocList_SelectedRows): void
+  {
+    if (selectedRows.DataRowCount == selectedRows.Length && selectedRows.DataRowCount > 0)
+      selectedRows.SelectAllSelectorControls!.each (function (this: JQuery) { this.checked = true; });
+  }
+
+  private static ClearSelectAllRowsSelector (selectedRows: BocList_SelectedRows): void
+  {
+    selectedRows!.SelectAllSelectorControls!.each (function (this: JQuery) { this.checked = false; });
+  }
+
+  //  Event handler for the selection selectorControl in the title row.
+  //  Applies the checked state of the title's selectorControl to all data rows' selectu=ion selectorControles.
+  //  bocList: The BocList to which the selectorControl belongs.
+  //  selectRowControlName: The name of the row selector controls.
+  //  isRowHighlightingEnabled: true to enable highting of the rows
+  public static OnSelectAllSelectorControlClick (bocList: HTMLElement, selectAllSelectorControl: JQuery, isRowHighlightingEnabled: boolean): void
+  {
+    var selectedRows = BocList._selectedRows[bocList.id]!;
+
+    if (selectedRows.Selection != BocList.rowSelectionMultiple)
+      return;
+    //  BocList_SelectRow will increment the length, therefor initialize it to zero.
+    if (selectAllSelectorControl.checked)
+      selectedRows.Length = 0;
+  
+    // TODO: Missing null check?
+    selectedRows.SelectRowSelectorControls!.each (function (this: JQuery)
+    {
+      var checkBox = this;
+  
+      var tableCell = checkBox.parentNode;
+      if (tableCell.nodeName !== 'TD')
+        throw 'Unexpected element type: \'' + tableCell.nodeName + '\'';
+  
+      var tableRow = tableCell.parentNode;
+      if (tableRow.nodeName !== 'TR')
+        throw 'Unexpected element type: \'' + tableRow.nodeName + '\'';
+  
+      var rowBlock = new BocList_RowBlock (tableRow, checkBox);
+      if (selectAllSelectorControl.checked)
+        BocList.SelectRow (bocList, rowBlock, isRowHighlightingEnabled);
+      else
+        BocList.UnselectRow (bocList, rowBlock, isRowHighlightingEnabled);
+    });
+    
+    if (! selectAllSelectorControl.checked)
+      selectedRows.Length = 0;
+  
+    selectedRows.OnSelectionChanged (bocList, false);
+  }
+
+  //  Event handler for the selection selectorControl in a data row.
+  //  Sets the _bocList_isSelectorControlClick flag.
+  private static OnRowSelectorClick(): void
+  {
+    BocList._isRowSelectorClickExecuting = true; //TODO either the documentation is incorrect or the body of the method
```

Comment:
Docs wrong. either way, the docs pretty much useless where and in the next function. not that doc-cleanup is a focus for the current task...